src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -344,6 +344,10 @@ CitusBeginModifyScan(CustomScanState *node, EState *estate, int eflags)
 		 * Given that pruning is deferred always for INSERTs, we get here
 		 * !EnableFastPathRouterPlanner  as well.
 		 */
+		if (jobQuery->commandType == CMD_INSERT)
+		{
+			currentPlan->fastPathRouterPlan = true;
+		}
```

Comment:
This might fix three cases given in #4091 too, looking into that.
See https://github.com/citusdata/citus/issues/4663#issuecomment-778114155